motoman_driver/MotoPlus/SimpleMessage.h

```diff
@@ -69,6 +69,10 @@ typedef enum
 	ROS_MSG_MOTO_WRITE_IO_GROUP = 2009,
 	ROS_MSG_MOTO_WRITE_IO_GROUP_REPLY = 2010,
 	ROS_MSG_MOTO_IOCTRL_REPLY = 2011,
+	ROS_MSG_MOTO_READ_MREGISTER = 2012,
+	ROS_MSG_MOTO_READ_MREGISTER_REPLY = 2013,
+	ROS_MSG_MOTO_WRITE_MREGISTER = 2014,
+	ROS_MSG_MOTO_WRITE_MREGISTER_REPLY = 2015,

```

Comment:
This will require a corresponding update to [REP-I4: Assigned Message Identifiers for the Simple Message Protocol](https://github.com/ros-industrial/rep/blob/bbd88b9b1dc9b61be5178c8b3a93b9bec488702a/rep-I0004.rst), specifically the *Vendor Specific Ranges: Motoman* section.

Reply:
Additionally: there should be no need for separate IDs for requests and replies. Requests and replies share the same ID, so `2012` would be read and `2013` would be write of M registers.

Reply:
I'm fine with removing the 'reply' message ID's. However, this is the method that is also used for motion-control messages.
https://github.com/ros-industrial/motoman/blob/kinetic-devel/motoman_driver/MotoPlus/SimpleMessage.h#L61

Reply:
> However, this is the method that is also used for motion-control messages.

I know, but that was also unnecessary.

I'm not sure why/how that got through the reviews at the time.

Reply:
I guess my only concern is breaking existing functionality.

There are users who are currently using the I/O interface, today. I assume that their code will be looking for the "_REPLY" message ID. If we remove these messages, will it break existing installations? (assuming they upgrade their MotoROS at some point)
https://github.com/ros-industrial/motoman/blob/kinetic-devel/motoman_driver/MotoPlus/SimpleMessage.h#L63-L71

Thoughts?

Reply:
> I assume that their code will be looking for the "_REPLY" message ID.

could you elaborate how? The numbers you refer to are numeric IDs of message structures. I would expect code to work with the numbers, but not necessarily (de)serialise to different classes/structures.

I'm not suggesting we remove the ones which are already there. It's unfortunate we have those, but removing them may break code outside of this repository.

I'm suggesting to not introduce *more* unnecessary ID allocations.

Reply:
This has been updated. Please note that I also changed the value for ROS_MSG_MOTO_WRITE_MREGISTER. So, that will need to be updated on the motomand-driver side too.

Reply:
Support for those things need to be added still, so there is nothing to update yet.